pkg/resource/training_job/custom_delta.go

```diff
@@ -60,3 +65,53 @@ func customSetDefaults(
 		}
 	}
 }
+
+// SM returns profiler related objects even if the user disables the profiler
+// customPostCompare detects if there is a diff
+func customPostCompare(latest *resource, desired *resource, delta *ackcompare.Delta) {
+	profilerConfigDiff := delta.DifferentAt("Spec.ProfilerConfig")
+	profilerRuleDiff := delta.DifferentAt("Spec.ProfilerRuleConfigurations")
+	if !profilerConfigDiff && !profilerRuleDiff {
+		return
+	}
+	profilerStatus := latest.ko.Status.ProfilingStatus
+	profilerDisabled := false
+
+	if ackcompare.IsNotNil(profilerStatus) {
+		//Do not remove profiler if user wants to enable it
+		if *profilerStatus == "Disabled" && !userInitiatesProfilerCheck(desired) {
+			profilerDisabled = true
+		} else {
+			return
+		}
+	} else {
+		return
+	}
+	// TODO: Replace remove delta with an ack version when its natively supported
+	if profilerConfigDiff && profilerDisabled {
+		removeDelta(delta, "Spec.ProfilerConfig")
+	}
+	if profilerRuleDiff && profilerDisabled {
+		removeDelta(delta, "Spec.ProfilerRuleConfigurations")
+	}
+}
+
+// userInitiatesProfilerCheck checks if the user enabled/re enabled the profiler.
+func userInitiatesProfilerCheck(desired *resource) bool {
+	profilerConfigPresent := ackcompare.IsNotNil(desired.ko.Spec.ProfilerConfig)
```

Comment:
can we just write the condition directly instead of creating vars, they are not reused